include/aws/mqtt/private/client_impl.h

```diff
@@ -85,77 +88,100 @@ struct aws_mqtt_client_connection {
 
     struct aws_mqtt_client *client;
 
-    /* The host information */
+    /* Channel handler information */
+    struct aws_channel_handler handler;
+    struct aws_channel_slot *slot;
+
+    /* The host information, changed by user when state is AWS_MQTT_CLIENT_STATE_DISCONNECTED */
     struct aws_string *host_name;
     uint16_t port;
     struct aws_tls_connection_options tls_options;
     struct aws_socket_options socket_options;
 
+    /* Connect parameters */
```

Comment:
All of those connect parameters are changed by user when the state is CONNECTED/DISCONNECTED, it will only be used by eventloop thread for next attempt of connecting, when the state is CONNECTING/RECONNECTING. So, we don't need a lock for them.